lightning/lightning.go

```diff
@@ -105,6 +145,59 @@ func (l *Lightning) run() error {
 }
 
 func (l *Lightning) Stop() {
+	if err := l.server.Shutdown(l.ctx); err != nil {
+		log.L().Warn("failed to shutdown HTTP server", log.ShortError(err))
+	}
 	l.shutdown()
-	l.wg.Wait()
+}
+
+func (l *Lightning) handleNewTask(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodGet:
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, len(l.taskCfgs))
+		return
+	case http.MethodPost:
+		break
+	default:
+		w.Header().Set("Allow", http.MethodGet+", "+http.MethodPost)
+		http.Error(w, "only GET and POST are allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	cfgs := l.taskCfgs
+	if cfgs == nil {
+		w.Header().Set("Cache-Control", "no-store")
+		http.Error(w, "server-mode not enabled", http.StatusNotImplemented)
+		return
+	}
+
+	data, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("cannot read request: %v", err), http.StatusBadRequest)
+		return
+	}
+	log.L().Debug("received task config", zap.ByteString("content", data))
+
+	cfg := config.NewConfig()
+	if err = cfg.LoadFromGlobal(l.globalCfg); err != nil {
+		http.Error(w, fmt.Sprintf("cannot restore from global config: %v", err), http.StatusInternalServerError)
+		return
+	}
+	if err = cfg.LoadFromTOML(data); err != nil {
+		http.Error(w, fmt.Sprintf("cannot parse task (must be TOML): %v", err), http.StatusBadRequest)
+		return
+	}
+	if err = cfg.Adjust(); err != nil {
+		http.Error(w, fmt.Sprintf("invalid task configuration: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	select {
+	case cfgs <- cfg:
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("task queued"))
+	default:
+		http.Error(w, "task queue overflowed, please try again later", http.StatusServiceUnavailable)
```

Comment:
The execution flow will reach here if the channel was closed, do we need to change the error message to adapt it. (Although we don't close the channel in the current implementation)